contracts/truffle/Migrations.sol

```diff
@@ -0,0 +1,23 @@
+pragma solidity >=0.4.21 <0.6.0;
+
+contract Migrations {
+  address public owner;
+  uint public last_completed_migration;
+
+  constructor() public {
+    owner = msg.sender;
+  }
+
+  modifier restricted() {
+    if (msg.sender == owner) _;
+  }
+
+  function setCompleted(uint completed) public restricted {
+    last_completed_migration = completed;
+  }
+
+  function upgrade(address new_address) public restricted {
+    Migrations upgraded = Migrations(new_address);
+    upgraded.setCompleted(last_completed_migration);
+  }
+}
```

Comment:
We should also follow the solidity guidelines and update documentation for this contract also.
I know we haven't followed this in other repositories. Since it's a new repo I believe it's a good starting point and then we can updated other repos  Migrations.sol as per guidelines 🤔

Reply:
I don't fully understand the methods of Migration.sol, we don't use this contract directly. Truffle internally uses it.  
 I referred to the documentation [here](https://truffleframework.com/docs/truffle/getting-started/running-migrations) but it doesn't explain the method. 

Do you have any references which can be used to document methods?

Reply:
It's strange that documentation is not updated on even on consensys repository.
https://github.com/ConsenSys/truffle-webpack-demo/blob/master/contracts/Migrations.sol

I found few links for reference:
https://ethereum.stackexchange.com/questions/8299/what-are-truffle-migrations
https://stackoverflow.com/questions/45205513/understanding-migration-script

May be we can for now update the Migration contract title and live with that until we find documentation for specific methods ❓

Reply:
`Migratins.sol` is auto-generated by Truffle. Truffle uses it to track which contracts have been deployed on the chain. Imagine a real DApp that uses on-chain contracts and then you add new contracts for new functionality. Truffle would see that (e.g. on Ethereum mainnet) and only deploy the new contracts.

Reply:
I have added a title to migration contract.